telemetry_core/src/main/java/com/newrelic/telemetry/metrics/SimpleMetricBatchSender.java

```diff
@@ -17,6 +17,12 @@
  */
 public class SimpleMetricBatchSender {
 
+  private static Function<Duration, HttpPoster> ctor = null;
```

Comment:
I really don't like adding static state here. I'd much prefer, if we want this class to support something other than okhttp, that we do that as a separate step. Can you remove this part from the PR, and we'll figure out a better way to do this as a follow-on?